apiserver/facades/client/bundle/bundle.go

```diff
@@ -152,9 +157,142 @@ func (b *BundleAPI) GetChanges(args params.BundleChangesParams) (params.BundleCh
 
 // ExportBundle exports the current model configuration as bundle.
 func (b *BundleAPI) ExportBundle() (params.StringResult, error) {
-	return params.StringResult{}, nil
+	var fail = func(failErr error) (params.StringResult, error) {
+		return params.StringResult{}, common.ServerError(failErr)
+	}
+
+	if err := b.checkCanRead(); err != nil {
+		return fail(err)
+	}
+
+	exportConfig := b.backend.GetExportConfig()
+	model, err := b.backend.ExportPartial(exportConfig)
+	if err != nil {
+		return fail(err)
+	}
+
+	// Fill it in charm.BundleData datastructure.
+	bundleData, err := b.fillBundleData(model)
+	if err != nil {
+		return fail(err)
+	}
+
+	bytes, err := yaml.Marshal(bundleData)
+	if err != nil {
+		return fail(err)
+	}
+
+	return params.StringResult{
+		Result: string(bytes),
+	}, nil
 }
 
 // Mask the new method from V1 API.
 // ExportBundle is not in V1 API.
 func (u *APIv1) ExportBundle() (_, _ struct{}) { return }
+
+// FillBundle fills the bundledata datastructure required for the exportBundle.
```

Comment:
Comment needs to match func name and is not necessary for unexported (starting with lower-case) functions unless they are complex. If you decide to keep the comment, however, 'data structure' might be missing a space?..